Some doctests are not compiled because they are required "derive" feature

After a long break I try to compile serde and I facing with a problem that some doc tests are not compiled on the current master (7b840897a9005d86273ca80e4881e34b5c77787c). I followed the steps in the `CONTRIBUTION.md` and this compile errors was really discouraging, because they are not mentioned, but it seems that currently they are expected, because deriving requires "derive" feature, which is not enabled by default.

<details><summary>cargo test output</summary>

```
d:\Projects\serde>cargo clean
d:\Projects\serde>cd serde
d:\Projects\serde\serde>cargo clean
d:\Projects\serde\serde>cargo test
   Compiling proc-macro2 v1.0.24
   Compiling unicode-xid v0.2.1
   Compiling syn v1.0.60
   Compiling serde_derive v1.0.126 (D:\Projects\serde\serde_derive)
   Compiling serde v1.0.126 (D:\Projects\serde\serde)
   Compiling quote v1.0.7
    Finished test [unoptimized + debuginfo] target(s) in 20.40s
     Running unittests (d:\Projects\serde\target\debug\deps\serde-2515a54deae22568.exe)

running 1 test
test ser::impls::test_format_u8 ... ok

test result: ok. 1 passed; 0 failed; 0 ignored; 0 measured; 0 filtered out; finished in 0.00s

   Doc-tests serde

running 71 tests
test src\de\mod.rs - de::DeserializeSeed (line 618) ... ok
test src\de\mod.rs - de::DeserializeOwned (line 579) ... ok
test src\de\mod.rs - de::DeserializeSeed (line 632) ... ok
test src\de\mod.rs - de::Deserializer::is_human_readable (line 1155) ... ok
test src\de\mod.rs - de::DeserializeSeed (line 665) ... ok
test src\de\mod.rs - de::Expected (line 433) ... ok
test src\de\ignored_any.rs - de::ignored_any::IgnoredAny (line 13) ... ok
test src\de\mod.rs - de::Error::custom (line 301) ... ok
test src\de\mod.rs - de::IntoDeserializer (line 2217) ... FAILED
test src\de\mod.rs - de::Expected (line 458) ... ok
test src\de\mod.rs - de::Unexpected (line 308) ... ok
test src\de\value.rs - de::value (line 4) ... FAILED
test src\de\mod.rs - de::VariantAccess::struct_variant (line 2157) ... ok
test src\de\mod.rs - de::VariantAccess::newtype_variant_seed (line 2054) ... ok
test src\de\mod.rs - de::VariantAccess::tuple_variant (line 2110) ... ok
test src\de\mod.rs - de::Visitor (line 1232) ... ok
test src\de\mod.rs - de::VariantAccess::unit_variant (line 2014) ... ok
test src\de\mod.rs - de::Visitor::expecting (line 1273) ... ok
test src\ser\fmt.rs - ser::fmt::&'a mut fmt::Formatter (line 20) ... FAILED
test src\integer128.rs - serde_if_integer128 (line 53) ... ok
test src\integer128.rs - serde_if_integer128 (line 12) ... ok
test src\integer128.rs - serde_if_integer128 (line 64) ... ok
test src\macros.rs - forward_to_deserialize_any (line 14) ... ok
test src\macros.rs - forward_to_deserialize_any (line 50) ... ok
test src\ser\impossible.rs - ser::impossible::Impossible (line 18) ... ok
test src\macros.rs - forward_to_deserialize_any (line 81) ... ok
test src\ser\mod.rs - ser::Serialize::serialize (line 222) ... ok
test src\ser\mod.rs - ser::Error::custom (line 192) ... ok
test src\ser\mod.rs - ser::SerializeMap (line 1723) ... ok
test src\ser\mod.rs - ser::SerializeStruct (line 1834) ... ok
test src\ser\mod.rs - ser::SerializeTuple (line 1520) ... ok
test src\ser\mod.rs - ser::SerializeSeq (line 1456) ... ok
test src\ser\mod.rs - ser::SerializeStructVariant (line 1894) ... ok
test src\ser\mod.rs - ser::SerializeTuple (line 1550) ... ok
test src\ser\mod.rs - ser::SerializeTupleVariant (line 1665) ... ok
test src\ser\mod.rs - ser::SerializeTupleStruct (line 1620) ... ok
test src\ser\mod.rs - ser::Serializer::collect_map (line 1294) ... ok
test src\ser\mod.rs - ser::Serializer::collect_seq (line 1257) ... ok
test src\ser\mod.rs - ser::Serializer::collect_str (line 1334) ... ok
test src\ser\mod.rs - ser::Serializer::is_human_readable (line 1407) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_bool (line 389) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_char (line 670) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_f32 (line 631) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_f64 (line 649) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_bytes (line 712) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_i16 (line 433) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_i128 (line 492) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_i32 (line 455) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_i8 (line 411) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_map (line 1131) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_i64 (line 473) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_newtype_struct (line 884) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_newtype_variant (line 912) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_none (line 741) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_some (line 774) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_seq (line 950) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_str (line 688) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_struct (line 1179) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_struct_variant (line 1215) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_tuple (line 995) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_tuple_struct (line 1053) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_tuple (line 1025) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_u128 (line 602) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_tuple_variant (line 1085) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_u16 (line 543) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_u32 (line 565) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_u8 (line 521) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_u64 (line 583) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_unit (line 809) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_unit_struct (line 829) ... ok
test src\ser\mod.rs - ser::Serializer::serialize_unit_variant (line 851) ... ok

failures:

---- src\de\mod.rs - de::IntoDeserializer (line 2217) stdout ----
error: cannot find derive macro `Deserialize` in this scope
 --> src\de\mod.rs:2222:10
  |
8 | #[derive(Deserialize)]
  |          ^^^^^^^^^^^

error[E0599]: no variant or associated item named `deserialize` found for enum `Setting` in the current scope
  --> src\de\mod.rs:2232:15
   |
9  | enum Setting {
   | ------------ variant or associated item `deserialize` not found here
...
18 |         Self::deserialize(s.into_deserializer())
   |               ^^^^^^^^^^^ variant or associated item not found in `Setting`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `deserialize`, perhaps you need to implement one of them:
           candidate #1: `Deserialize`
           candidate #2: `DeserializeSeed`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0599`.
Couldn't compile the test.
---- src\de\value.rs - de::value (line 4) stdout ----
error: cannot find derive macro `Deserialize` in this scope
 --> src\de\value.rs:9:10
  |
8 | #[derive(Deserialize)]
  |          ^^^^^^^^^^^

error[E0599]: no variant or associated item named `deserialize` found for enum `Setting` in the current scope
  --> src\de\value.rs:19:15
   |
9  | enum Setting {
   | ------------ variant or associated item `deserialize` not found here
...
18 |         Self::deserialize(s.into_deserializer())
   |               ^^^^^^^^^^^ variant or associated item not found in `Setting`
   |
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following traits define an item `deserialize`, perhaps you need to implement one of them:
           candidate #1: `Deserialize`
           candidate #2: `DeserializeSeed`

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0599`.
Couldn't compile the test.
---- src\ser\fmt.rs - ser::fmt::&'a mut fmt::Formatter (line 20) stdout ----
error: cannot find derive macro `Serialize` in this scope
 --> src\ser\fmt.rs:24:10
  |
7 | #[derive(Serialize)]
  |          ^^^^^^^^^

error: cannot find attribute `serde` in this scope
 --> src\ser\fmt.rs:25:3
  |
8 | #[serde(rename_all = "kebab-case")]
  |   ^^^^^

error[E0599]: the method `serialize` exists for reference `&MessageType`, but its trait bounds were not satisfied
  --> src\ser\fmt.rs:33:14
   |
9  | pub enum MessageType {
   | -------------------- doesn't satisfy `MessageType: Serialize`
...
16 |         self.serialize(f)
   |              ^^^^^^^^^ method cannot be called on `&MessageType` due to unsatisfied trait bounds
   |
   = note: the following trait bounds were not satisfied:
           `MessageType: Serialize`
           which is required by `&MessageType: Serialize`
   = help: items from traits can only be used if the trait is implemented and in scope
   = note: the following trait defines an item `serialize`, perhaps you need to implement it:
           candidate #1: `Serialize`

error: aborting due to 3 previous errors

For more information about this error, try `rustc --explain E0599`.
Couldn't compile the test.

failures:
    src\de\mod.rs - de::IntoDeserializer (line 2217)
    src\de\value.rs - de::value (line 4)
    src\ser\fmt.rs - ser::fmt::&'a mut fmt::Formatter (line 20)

test result: FAILED. 68 passed; 3 failed; 0 ignored; 0 measured; 0 filtered out; finished in 9.62s

error: test failed, to rerun pass '--doc'

d:\Projects\serde\serde>
```
</details>

If you change command to `cargo test --features derive`, then compilation will be successful

--- a/CONTRIBUTING.md
+++ b/CONTRIBUTING.md
@@ -31,7 +31,7 @@ tests for you.
 
 ```sh
 # Test all the example code in Serde documentation
-cargo test
+cargo test --features derive
 ```
 
 ##### In the [`test_suite/deps`] directory

